src/Core/Domain/Attachment/Command/EditAttachmentCommand.php

```diff
@@ -85,15 +85,15 @@ public function getAttachmentId(): AttachmentId
     }
 
     /**
-     * @return string|null
+     * @return string
      */
     public function getPathName(): ?string
     {
         return $this->pathName;
     }
 
     /**
-     * @param string|null $pathName
+     * @param string $pathName
```

Comment:
Same here, all of these fields are null when used in a partial edit usecase. The phpDoc is right. If PHPStan marks these statements as wrong, something is wrong either in our PHPStan configuration or in PHPStan internal logic.

Reply:
Wrong ! The data passed here come from a SplFileInfo object and `getPathname` method return always a string.

Reply:
Nothing into the implementation of `EditAttachmentCommand` obliges the user to get this data from a SplFileInfo object. Our usage of `EditAttachmentCommand` class inside `AttachmentFormDataHandler` uses SplFileInfo data to fill it, but we could fetch the data from another source later.

Moreover we must let modules and shop developers free to choose where they pick the data to fill in `EditAttachmentCommand`. This code must be built to allow extension of the behavior by module and shop developers.

Reply:
> Wrong !

Please dont use exclamation marks in your comments 😅 
You probably dont want it to sound that way but when I read it, in my mind I think of this:
<img src="https://lesmauxdevente.fr/wp-content/uploads/2018/07/Rien-ne-vaut-une-objection.png">

It feels a bit aggressive 😅